source/apiVolontaria/volunteer/views.py

```diff
@@ -342,6 +345,76 @@ def get_queryset(self):
             queryset = queryset.filter(user__username=username)
         return queryset
 
+    def generate_email_invitation(self, obj):
+        cal = Calendar()
+
+        cal.add('prodid', '-//{0} Events Calendar//'.format(obj.event.task_type.name, ))
+        cal.add('version', '2.0')
+
+        ical_event = ICalEvent()
+        ical_event.add('summary', obj.event.task_type.name)
+        ical_event.add('dtstart', obj.event.start_date)
+        ical_event.add('dtend', obj.event.end_date)
+        ical_event.add('dtstamp', obj.event.end_date)
+        ical_event['uid'] = str(obj.id)
+
+        cal.add_component(ical_event)
+        calendar = cal.to_ical()
```

Comment:
All the code used to create the `.ics` can be put in a generic file of function so we can use it in other place of the app and the view is more easy to understand.

Reply:
I agree, I suggest to place it in the Participation model.

Reply:
Can be a good place to start yes.